src/metabase/driver/firebolt.clj

```diff
@@ -101,23 +101,35 @@
     :timestamp     :type/DateTime
     :timestamptz   :type/DateTimeWithLocalTZ
     :varchar       :type/Text
+    :struct        :type/Text
     (keyword "timestamp with timezone")    :type/DateTime
     (keyword "timestamp without timezone") :type/DateTime})
 
 ; Map firebolt data types to base types
 (defmethod sql-jdbc.sync/database-type->base-type :firebolt [_ database-type]
    (database-type->base-type database-type))
 
-; Concatenate the elements of an array based on array elemets type (coverting array data type to string type to apply filter on array data)
-(defn is-string-array? [os]
-  (if (= (type (first (vec os))) java.lang.String) (str "['" (clojure.string/join "','" os) "']") (str "[" (clojure.string/join "," os) "]")))
-
-; Handle array data type
+(defn array-to-string [os]
+  (let [convert-element (fn [element]
+  (do (log/info "Class: " (.getName (.getClass element)) "is array" (.isArray (class element)))
+                          (cond
+                            (.isArray (class element))
+                            (array-to-string (into [] element)) ; Handle nested arrays
+                            (instance? java.sql.Timestamp element)
+                            (.format (java.text.SimpleDateFormat. "yyyy-MM-dd HH:mm:ss") element)
+                            (instance? java.lang.String element)
+                            (str "'" element "'")
```

Comment:
do we support other types that require processing, like bytea? Also, how are other date types handled? Do we need another format here for `date` and `timestamptz`?